### ISSUE TITLE
missing instruction how to find where VS Code extensions are installed to be able to include them into the path

### ISSUE THREAD

**Issue by aisbergde** [MicrosoftDocs/azure-dev-docs#34]
your instruction:
"verify that the folder where you installed the Azure Functions Core Tools is included in your PATH environment variable"
installation of extensions happens without asking the user where they are installed and there is no easy or intuitive way to get this information

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: 6a6a3f99-4725-7f93-301e-da4ba69d4986
* Version Independent ID: 3477aa24-7135-e6e9-3791-17d66052589e
* Content: [Tutorial: Create and deploy serverless Azure Functions in Python with Visual Studio Code](https://docs.microsoft.com/en-us/azure/python/tutorial-vs-code-serverless-python-01)
* Content Source: [python/tutorial-vs-code-serverless-python-01.md](https://github.com/MicrosoftDocs/azure-dev-docs/blob/master/python/tutorial-vs-code-serverless-python-01.md)
* Product: **azure-python**
* GitHub Login: @kraigb
* Microsoft Alias: **kraigb**

**Comment by aisbergde** [comment]
Also when I try to find the place and find something like this and add this to the path, the "func" will not work:
C:\Users\ggoer\.vscode\extensions\ms-azuretools.vscode-azurefunctions-0.20.0
Please add a better description:
- where to find the path
- what exactly should be the path (maybe a subfolder of the extension folder)
- what to do after adding the folder to the path

The error I get doesn't help me to solve it:
func : Die Benennung "func" wurde nicht als Name eines Cmdlet, einer Funktion, einer Skriptdatei oder eines ausführbaren Programms erkannt. Überprüfen Sie die Schreibweise des Namens, oder ob der Pfad korrekt ist (sofern 
enthalten), und wiederholen Sie den Vorgang.

**Comment by tsilver** [comment]
I am having the same problem.  
`func
func : The term 'func' is not recognized as the name of a cmdlet, function, script file, or operable program. Check the 
spelling of the name, or if a path was included, verify that the path is correct and try again.
At line:1 char:1
+ func
+ ~~~~
    + CategoryInfo          : ObjectNotFound: (func:String) [], CommandNotFoundException
    + FullyQualifiedErrorId : CommandNotFoundException`

**Comment by jayvoy** [comment]
Same problem: 'func' is not recognized as an internal or external command, operable program or batch file

**Comment by tsilver** [comment]
Found it here on my machine: C:\Users\Tom\AppData\Roaming\npm\node_modules\azure-functions-core-tools\bin

**Comment by tsilver** [comment]
> Same problem: 'func' is not recognized as an internal or external command, operable program or batch file

C:\Users\Tom\AppData\Roaming\npm\node_modules\azure-functions-core-tools\bin

**Comment by jayvoy** [comment]
It took 2 hours but I've solved the "func" error message by downloading 3 things:

1) .NET framework, 2) PowerShell, 3) node.js/NPM, and I believe one other dependency required from this list before it would install.  
If I remember correctly, I launched the node.js/NPM install from by clicking a file that opened powershell. The node.js/NPM piece then also installed Chocolatey though it gave a 'fail' message somewhere in the long string of text it ouput.  It also installed Python 2.7 and MS VS Code then changed the interpreter to 2.7 so that  had to be changed back.  

fwiw, all those terms above are completely new to me.  I don't know what any of them mean or what they do.  It would be nice for a simple python programmer to abstract away as much of this as possible going forward.

Am I missing something though?  Is there an easier way to install all these pieces?  I am on Windows 7 - perhaps some of this is due to my old OS?

**Comment by jayvoy** [comment]
I spoke too soon - the node.js piece appears not to have installed correctly and now I get an ECONNREFUSED error when trying to "Attach to Python Functions" a couple steps ahead of this.

**Comment by kraigb** [comment]
Thanks for all the info, everyone. I completely agree that having to install .NET Core and npm to use Functions with Python is a bother; I've stated that a number of times to the engineering team, but there's no estimate for when it might work more simply. For now, the instructions on https://docs.microsoft.com/en-us/azure/azure-functions/functions-run-local#v2 are accurate (but note that Chocolately is a different packaging systems, and unless you're already using it, there's no need to use it here). If you don't want to bother with .NET Core, then there is the [extension bundles](https://docs.microsoft.com/en-us/azure/azure-functions/functions-bindings-register#extension-bundles) approach, but I don't think it's necessarily easier.

In any case, when you run the 'npm install -g azure-functions-core-tools' command, the core tools package is installed in the same folder that contains the npm command, which is the same as the install folder for Node.js itself. You should be able to find that folder by using the command `where npm.*`. The folder is also displayed at the end of the output of the `npm -l` command.

That particular node.js folder should be on the PATH after installing Node.js (following https://docs.npmjs.com/getting-started/installing-node#osx-or-windows). If it isn't, then running npm shouldn't work in the first place. Still, you can check this by looking at the PATH settings in Control Panel's environment variables section.

When you run `npm install -g azure-functions-core-tools`, you should then have a func.cmd file in that same node.js folder. Running `where func.cmd` should show that same folder. Given that this node.js folder should already be in your PATH, then the func command should work. (@jayvoy with the secondary problem you encountered, please file a separate issue for this to keep this present thread focused on the installation matters).

Bottom line is that with npm on the PATH and using the -g switch, then func.cmd should also be on the PATH automatically. If it's not, then there are a couple of possibilities:

1. Make sure you use the -g switch with npm install, which means "install globally into the node.js install folder". If you don't include -g, npm installs the core tools (and func.cmd) in the current folder only, which won't be found if you attempt to run `func` from a different folder.

2. Examine the output of the npm install command. If an error occurs, then the package didn't install and then `func` won't work.

 Let me know if you're still having difficulties.

**Comment by aisbergde** [comment]
I gave up. And I rejected the project.
The idea was simply to load some reports from reports.office.com into an Azure data base. But it looks it is so complicated to load something from one Microsoft product into another:
Find out how this OAuth works in Azure and Microsoft products
Understand which "scenario" is required
Python to use OAuth (because I used this in another project for OAuth2 access to load data via  REST Api and this was _not_ complicated)
Azure Function to run this Python
Getting VS Code to be able to use Python in an Azure Function.

Many time is required to investigate how to simply load data from one Azure product into another. There is so much documentation, but important points are missing in the docs and it takes too much time to find a solution. Maybe other people can. I am a stupid old white man, a database architect and developer, and I lost a little bit in all this complicated ways how to move between Azure cloud products. Too much diversity in ways in products...

**Comment by jayvoy** [comment]
Hi Kraig - Thank you for absorbing our frustration and offering helpful tips for those who kept trying.

I can confirm that I've now made it through the "Deploy to Azure" step (about 4 steps beyond this page) and have successfully worked the example up to that point.

If I may, I have a few suggestions for the "Prerequisites" section to make it more complete.  Perhaps consider including a note about needing 1) node.js, 2) npm and 3) .NET Core in addition to those prereq's already listed.  Your link in the comment above was helpful to install the node.js/npm piece as well as the command line instruction to then install 'azure function core tools'

An important note on the node.js/npm install for future readers (if you follow the link in Kraig's comment to download).  At some point in the installation, there will be a checkbox for "Do you want to automatically ...." do such and such.  Do NOT check that box.  I did and it installed Chocolatey (not needed as per Kraig's comments), Python 2.7 (why?), and a whole boat load of other things.  Perhaps more importantly, the demo wouldn't work.  I performed a system restore to clear it all off and started again.  

It's taken about 4 days but I can finally run "Hello Jason" 

Thank you for the documentation, for monitoring these threads, and for being patient with us along during our moments of frustration.

**Comment by kraigb** [comment]
@jayvoy That's a great suggestion to be more specific about options in the installer. I'll plan to go through the process again and double-check those details and what, exactly, gets installed when and where.

**Comment by kraigb** [comment]
I've made a PR to the docs to clarity the prereqs for core tools and install options, which should smooth the processes. Accordingly, I'm closing the issue.

@aisbergde With regards to your most recently comments, I encourage you to submit overall usability issues to the repo for Azure Functions itself, because what you raise is beyond the scope of docs (which is the scope of this present repo), and can't really be fixed properly in docs. You can file issues on https://github.com/Azure/Azure-Functions/issues. The engineering team will love to hear of your difficulties so they can try to improve the overall flow.

**Comment by aisbergde** [comment]
@kraigb of course you are right. This is the wrong place for my comments. I was a little bit disappointed. But on the other side I am excited that currently user feedback is "used" by Microsoft to improve product and documentation quality. Some years ago I spent some time on feedback and it was "lost in... I don't know where". But currently I feel "listened" and my motivation to give feedback is much higher. This is the right way!

**Comment by kraigb** [comment]
You're very welcome. We're encouraged in the docs org to monitor and respond to feedback; the degree of success in such efforts varies, as you might expect, but we certainly do care about what you have to say as do many of the engineering teams who have their work on GitHub.

Thanks again for taking the time to file issues and converse with us. It's much appreciated, because on my end you sometimes wonder if anyone is actually reading the docs. :)
